cpp/src/arrow/flight/sql/CMakeLists.txt

```diff
@@ -66,36 +66,34 @@ else()
                                       ${ARROW_TEST_LINK_LIBS})
 endif()
 
-add_arrow_test(flight_sql_test
-               SOURCES
-               client_test.cc
-               server_test.cc
-               example/sqlite_sql_info.cc
-               STATIC_LINK_LIBS
-               ${ARROW_FLIGHT_SQL_TEST_LINK_LIBS}
-               LABELS
-               "arrow_flight_sql")
-
 # Build test server for unit tests
 if(ARROW_BUILD_TESTS OR ARROW_BUILD_EXAMPLES)
   find_package(SQLite3Alt REQUIRED)
 
-  add_executable(flight_sql_test_server
-                 test_server_cli.cc
-                 example/sqlite_sql_info.cc
-                 example/sqlite_statement.cc
-                 example/sqlite_statement_batch_reader.cc
-                 example/sqlite_server.cc
-                 example/sqlite_tables_schema_batch_reader.cc)
+  set(flight_sql_test_server_SRCS
```

Comment:
Two things:

This should perhaps be something like `ARROW_FLIGHT_SQL_TEST_SERVER_SRCS`, and why does the CLI need anything more than `sqlite_sql_info.cc`?

Reply:
Renamed!
Now that the sqlite server runs in the same process as the tests on `server_test.cc`, it needs to build the same files as the CLI

Reply:
Ah, my bad, somehow I got mixed up and thought the server sources were getting built into the client CLI. This looks fine.